### PR TITLE
added IaC for the cicd-for-data-processing-workflow

### PR DESCRIPTION
**Pull request by parasmamgain**
Hi,
Following are the changes added
- Iac Code using terraform that follows the cicd-dataprocessing-workflow as described using gcloud and the blogpost
- readme file updating about the instructions on how to use this IaC Code

### CODE REVIEW

File: example/1.terraform-automation/variables.tf [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+variable "composer_dag_name_test" {
+  type        = string
+  default     = "test_word_count"
+  description = "The Composer DAG name(for test) to be passed as environment variable."
+
+}
+
+variable "image_version" {
+  type        = string
+  description = "The version of the aiflow running in the cloud composer environment."
```
Comment (kingman) [inline example/1.terraform-automation/variables.tf]: airflow

File: example/1.terraform-automation/README.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+## Execution
+
+1. git clone this repository and `cd ci-cd-for-data-processing-workflow/example/1.terraform-automation`.
+2. Perform a gcloud login using `gcloud auth application-default login`
+3. Update `terraform.tfvars` with the values.
+4. Execute `terraform init`
+5. Execute `terraform plan` and validate the resources displayed in the output.
+6. Execute `terraform apply` and confirm with `yes` when asked to create resources in your google project.
+7. Once the above steps complete, you would have created the GCP resources listed in the `Introduction` section. You should now also be able to see the two source code repositories created in your project, one for the `terraform-automation-source` and another for the `data-pipeline-source`.
+8. You can now push the three folders(build-pipeline, data-processing-code, workflow-dag) present inside the `source-code` folder ([link](https://github.com/parasmamgain/ci-cd-for-data-processing-workflow/tree/master/source-code)) in the Code Source repository created with the name `data-pipeline-source`.
```
Comment (kingman) [inline example/1.terraform-automation/README.md]: pls, change the reference the official repo

File: example/1.terraform-automation/README.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+- Enable Google API's
+
+## Pre requistie
+
+1. You must have a GCP project created with `project owner` permissions.
+2. Terraform and Gcloud cli are installed in your machine.
+
+## Execution
+
+1. git clone this repository and `cd ci-cd-for-data-processing-workflow/example/1.terraform-automation`.
```
Comment (kingman) [inline example/1.terraform-automation/README.md:23]: explicitly give the reader the complete git clone command, making it easier to follow alone the guide

File: example/1.terraform-automation/README.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+2. Perform a gcloud login using `gcloud auth application-default login`
+3. Update `terraform.tfvars` with the values.
```
Comment (kingman) [inline example/1.terraform-automation/README.md]: explicitly spell out what need to be updated, it looks like project id is that only thing that need to be provided, everything else have default values. if it's case, state is here

File: example/1.terraform-automation/README.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+- Cloud Build Trigger
+- Cloud Source Repository
+- Service Account for Cloud Composer
+- Pub-Sub respource
+- Network resources like VPC and Subnet
+- Enable Google API's
+
+## Pre requistie
+
+1. You must have a GCP project created with `project owner` permissions.
```
Comment (kingman) [inline example/1.terraform-automation/README.md:18]: Could we suggest reader to use cloud-shell as the default option, as tool is be default installed here.

File: example/1.terraform-automation/main.tf [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  result_bucket_prod           = "${var.project_id}-composer-result-prod-tf"
+  dataflow_staging_bucket_prod = "${var.project_id}-dataflow-staging-prod-tf"
+  project_number               = data.google_project.project.number
+}
+
+data "google_project" "project" {
+  project_id = var.project_id
+}
+
+resource "google_project_service" "project" {
```
Comment (kingman) [inline example/1.terraform-automation/main.tf]: consider use terraform-google-modules/project-factory/google//modules/project_services to enable services

**Review by kingman — COMMENTED** [review]
Hi, left few comment. Please have a look. 

Best regards,
Charlie

**Comment by parasmamgain** [comment]
Hi @kingman ,

thanks for reviewing the above. Please feel free to review the above with the updated code.

thanks & regards,
Paras Mamgain

**Pull request merged**